darts/models/forecasting/regression_model.py

```diff
@@ -809,6 +827,17 @@ def fit(
         ):
             logger.warning("Provided `n_jobs_multioutput_wrapper` wasn't used.")
 
+        # Checking for correct values the provided sample weights
+        if isinstance(sample_weight, str):
+            raise_if(
+                sample_weight not in ["equal", "linear_decay", "exponential_decay"],
+                f"Invalid value for `sample_weight`: {sample_weight}."
+                f"Possible values are: equal, linear_decay, exponential_decay.",
+            )
+        elif isinstance(sample_weight, TimeSeries):
+            # The error is caught later, should we still verify it here?
```

Comment:
why this is an error if in the docstring you are saying that `If a TimeSeries is passed, then those weights are used.`?

Reply:
This is a placeholder, we are trying to think if it's worth it to check that the time index of the passed series matches the index of the training series for example.